lib/bigtable/filter.js

```diff
@@ -0,0 +1,299 @@
+/*!
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /*!
+ * @module bigtable/filter
+ */
+
+'use strict';
+
+var createErrorClass = require('create-error-class');
+var is = require('is');
+var Entry = require('./entry.js');
+
+var UnknownFilter = createErrorClass('UnknownFilter', function(filterName) {
+  this.message = 'Unknown filter "' + filterName + '"';
+});
+
+/**
+ *
+ */
+function Filter() {
+  this.filters_ = [];
+}
+
+/**
+ *
+ */
+Filter.createRange = function(key, range) {
+  var data = {};
+
+  Object.keys(range).forEach(function(boundName) {
+    var bound = range[boundName];
+
+    if (!is.object(bound)) {
+      bound = {
+        value: bound || ''
+      };
+    }
+
+    var boundKey = boundName + key;
+    var valueKey = boundKey + (bound.inclusive ? 'Inclusive' : 'Exclusive');
+
+    data[boundKey] = {};
+    data[boundKey][valueKey] = Entry.convertToBytes(bound.value);
+  });
+
+  return data;
+};
+
+/**
+ *
+ */
+Filter.convertToRegExpString = function(regex) {
+  if (is.regexp(regex)) {
+    return regex.toString();
+  }
+
+  if (is.array(regex)) {
+    return '(' + regex.join('|') + ')';
```

Comment:
We could, do you think that would be useful? I think my intended use for arrays was where you didn't want to create a RegExp to search for multi value - e.g. instead of making them type `/(a|b|c)/` they could just do `['a', 'b', 'c']`.

Although, this actually brings up a bigger issue (which I kinda overlooked) in that we need a way to document Filters clearly, because they're kind of a beast and since we decided to go POJO on this one, I'm not sure where an ideal place to document them would be.